pybamm/solvers/base_solver.py

```diff
@@ -1160,11 +1160,34 @@ def step(
             raise pybamm.SolverError(
                 f"Step time must be at least {pybamm.TimerTime(step_start_offset)}"
             )
-
         t_start = old_solution.t[-1]
-        t_end = t_start + dt
-        # Calculate t_eval
-        t_eval = np.linspace(t_start, t_end, npts)
+        t_end = t_start
+        t_eval = np.array([t_start])
+
+        if non_linear_time is not None:
+            # Checking if non_linear_time lies within range
+            if non_linear_time[0] != 0 or non_linear_time[-1] != dt:
+                raise pybamm.SolverError(
+                    "Elements inside array non_linear_time must lie in the closed interval 0 to dt"
+                )
+
+            for i in range(len(non_linear_time) - 1):
```

Comment:
cany use `any` and remove the loop, then just make `t_eval = non_linear_time + t_start`

Reply:
So should I not check if difference between elements is larger than `pybamm.TimerTime(step_start_offset)`?

Reply:
I get it now. But wouldn't we still need a `time_diff` variable to store time difference so that we can increment `t_end` by that much?

Reply:
you don't need to do this check. we just need `dt` to be bigger than `step_start_offset`